src/Commands/D9ify/ProcessCommand.php

```diff
@@ -0,0 +1,694 @@
+    ];
+
+
+    /**
+     * @var Directory
+     */
+    protected Directory $sourceDirectory;
+    /**
+     * @var Directory
+     */
+    protected Directory $destinationDirectory;
+
+    /**
+     * Clone a pantheon site and spelunk the contents to create new D9 site.
+     *
+     * @authorize
+     *
+     * @command d9ify:process
+     *
+     * @aliases d9p
+     *
+     * @param string $sourceSite
+     *   Pantheon Site name/ID.
+     * @param ?string $destinationSite
+     *   Pantheon Destination Site (optional)
+     *
+     * @return void
+     * @throws \JsonException
+     *
+     * @usage terminus d9ify:process {sourceSiteName}
+     * @usage terminus d9ify:process {sourceSiteName} {destinationSiteName}
+     */
+    public function process($sourceSite, $destinationSite = null)
+    {
+        $this->output()->writeln(static::$HELP_TEXT);
+        try {
+            // Handle Source Site.
+            if ($sourceSite instanceof Site) {
+                $sourceSiteObject = $sourceSite;
+            }
+            if (is_string($sourceSite)) {
+                $sourceSiteObject = $this->getSite($sourceSite);
+            }
+            if (isset($sourceSiteObject) && $sourceSiteObject instanceof Site) {
+                $this->getContainer()->add("sourceDir", Directory::class)
+                    ->addArgument(['site' => $sourceSiteObject]);
+                $this->sourceDirectory = $this->getContainer()->get("sourceDir");
+            }
+
+            // Create destination Site if not exists or value is null.
+            if ($destinationSite == null) {
+                $destinationSite = $sourceSiteObject->getName() . "-" . date("Y");
+            }
+            if ($destinationSite instanceof Site) {
+                $destinationSiteObject = $destinationSite;
+            }
+            if (is_string($destinationSite)) {
+                $destinationSiteObject = $this->sites()->get($destinationSite);
+            }
+            \Kint::dump($this->sites());
+            exit();
+            if (!isset($destinationSiteObject) || !$destinationSiteObject instanceof Site) {
+                $this->getContainer()->add(CreateCommand::class);
+                $createCommand = $this->getContainer()->get(CreateCommand::class);
+                $createCommand->create(
+                    $destinationSite,
+                    $destinationSite,
+                    'drupal9',
+                    ['org' => null]
+                );
+                $destinationSiteObject = $this->getSite($destinationSite);
+            }
+            if (isset($destinationSiteObject) || $destinationSiteObject instanceof Site) {
+                $this->getContainer()->add('destinationDir', Directory::class)
+                    ->addArgument(['site' => $destinationSiteObject ]);
+                $this->destinationDirectory = $this->getContainer()->get('destinationDir');
+            }
+
+
```

Comment:
@TeslaDethray when I get here, $sourceSiteObject and $destinationSiteObject are the same site and I'm not sure why.